### PR TITLE
Deprecate elgg instanceof

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: docs/guides/notifications.rst [hunk 1]
```diff
@@ -286,7 +286,7 @@ Example:
 	function discussion_get_subscriptions($hook, $type, $subscriptions, $params) {
 		$reply = $params['event']->getObject();
 
-		if (!elgg_instanceof($reply, 'object', 'discussion_reply')) {
+		if (!$reply instanceof \ElggDiscussionReply) {
```
Comment (jeabakker) [inline docs/guides/notifications.rst]: this has been replaced by ElggComments

File: docs/tutorials/blog.rst [hunk 2]
```diff
@@ -273,7 +273,7 @@ the entity should of course be a my_blog post. Add this function to your
 
     function my_blog_set_url($hook, $type, $url, $params) {
         $entity = $params['entity'];
-        if (elgg_instanceof($entity, 'object', 'my_blog')) {
+        if ($entity instanceof \ElggBlog) {
```
Comment (jeabakker) [inline docs/tutorials/blog.rst]: Is this registered in the tutorial?

File: engine/lib/deprecated-3.1.php [hunk 3]
```diff
@@ -142,3 +142,35 @@ function elgg_load_css($name) {
...
+ * @param string $type    Entity type
+ * @param string $subtype Entity subtype
+ *
+ * @return bool
+ * @since 1.8.0
+ *
+ * @deprecated 3.1
+ */
+function elgg_instanceof($entity, $type = null, $subtype = null) {
+	elgg_deprecated_notice(__METHOD__ . ' is deprecated.', '3.1');
```
Comment (jeabakker) [inline engine/lib/deprecated-3.1.php]: ... use PHP instanceof

File: docs/guides/notifications.rst (hunk 1 above)
Comment (jdalsem) [reply docs/guides/notifications.rst]: Its just an example

File: engine/classes/Elgg/Database/EntityTable.php [hunk 4]
```diff
@@ -390,7 +390,7 @@ public function get($guid, $type = null, $subtype = null) {
 		$guid = (int) $guid;
 
 		$entity = $this->getFromCache($guid);
-		if ($entity && elgg_instanceof($entity, $type, $subtype)) {
+		if ($entity instanceof \ElggEntity && $entity->getType() === $type && $entity->getSubtype() === $subtype) {
```
Comment (jdalsem) [inline engine/classes/Elgg/Database/EntityTable.php]: this line is causing the erros... need to check type/subtype differently

also add integration tests as only a weird simpletest caught this

**Pull request merged**